src/uu/uname/src/uname.rs

```diff
@@ -15,92 +15,156 @@ use uucore::{
-    pub static NODENAME: &str = "nodename";
-    pub static KERNEL_VERSION: &str = "kernel-version";
-    pub static KERNEL_RELEASE: &str = "kernel-release";
-    pub static MACHINE: &str = "machine";
-    pub static PROCESSOR: &str = "processor";
-    pub static HARDWARE_PLATFORM: &str = "hardware-platform";
-    pub static OS: &str = "operating-system";
+static ALL: &str = "all";
+static KERNEL_NAME: &str = "kernel-name";
+static NODENAME: &str = "nodename";
+static KERNEL_VERSION: &str = "kernel-version";
+static KERNEL_RELEASE: &str = "kernel-release";
+static MACHINE: &str = "machine";
+static PROCESSOR: &str = "processor";
+static HARDWARE_PLATFORM: &str = "hardware-platform";
+static OS: &str = "operating-system";
+
+pub struct UNameOutput {
+    pub kernel_name: Option<String>,
+    pub nodename: Option<String>,
+    pub kernel_release: Option<String>,
+    pub kernel_version: Option<String>,
+    pub machine: Option<String>,
+    pub os: Option<String>,
+    pub processor: Option<String>,
+    pub hardware_platform: Option<String>,
 }
 
-#[uucore::main]
-pub fn uumain(args: impl uucore::Args) -> UResult<()> {
-    let matches = uu_app().try_get_matches_from(args)?;
-
-    let uname = PlatformInfo::new().map_err(|_e| USimpleError::new(1, "cannot get system name"))?;
-
-    let mut output = String::new();
-
-    let all = matches.get_flag(options::ALL);
-    let kernel_name = matches.get_flag(options::KERNEL_NAME);
-    let nodename = matches.get_flag(options::NODENAME);
-    let kernel_release = matches.get_flag(options::KERNEL_RELEASE);
-    let kernel_version = matches.get_flag(options::KERNEL_VERSION);
-    let machine = matches.get_flag(options::MACHINE);
-    let processor = matches.get_flag(options::PROCESSOR);
-    let hardware_platform = matches.get_flag(options::HARDWARE_PLATFORM);
-    let os = matches.get_flag(options::OS);
-
-    let none = !(all
-        || kernel_name
-        || nodename
-        || kernel_release
-        || kernel_version
-        || machine
-        || os
-        || processor
-        || hardware_platform);
-
-    if kernel_name || all || none {
-        output.push_str(&uname.sysname().to_string_lossy());
-        output.push(' ');
-    }
-
-    if nodename || all {
-        output.push_str(&uname.nodename().to_string_lossy());
-        output.push(' ');
+impl UNameOutput {
+    fn display(&self) -> String {
+        let mut output = String::new();
+        [
+            self.kernel_name.as_ref(),
+            self.nodename.as_ref(),
+            self.kernel_release.as_ref(),
+            self.kernel_version.as_ref(),
+            self.machine.as_ref(),
+            self.os.as_ref(),
+            self.processor.as_ref(),
+            self.hardware_platform.as_ref(),
+        ]
+        .iter()
+        .for_each(|name| {
+            if let Some(name) = name {
+                output.push_str(name);
+                output.push(' ');
+            }
+        });
```

Comment:
This looks like it could be done with `.flatten`. So something like:
```rust
for name in [...].into_iter().flatten() {
    output.push_str(name);
    output.push(' ');
}
```

Reply:
TIL; didnt know you could flatten an option. Pretty cool, thx!